docs/sec-blueprint/08-Hardening.md

```diff
@@ -41,16 +41,193 @@ invalid binaries on the device, and to prevent copying of security
 related data from the device.  
 There are three main areas of focus for hardening an embedded device:
 
+**Boot Hardening**: Steps/requirements to configure the boot sequence,
+in order to restrict the device from executing anything other than the
+approved software image.
+
+**System Hardening**: Best practices associated with the configuration
+of an embedded Linux based operating system. This section includes both
+hardening of the kernel itself, as well as specific configurations and
+patches used to protect against known vulnerabilities within the build
+and configuration of the root filesystem.
+
+**Application Hardening:** Best practices to apply to the build and
+release of user space applications, in order to reduce the number of
+attack surfaces used by potential attackers.
+
+Secure Boot Software Flow Steps
+-------------------------------
+
+1.  After power on, the secure processor will perform the verification
+    of the Stage 1 boot image, the stage 2 boot image and the Secure
+    loader image.
+
+    a.  If any of the images fail the verification process the device
+        will restart.
+
+2.  Upon successful verification of all of the boot and loader images,
+    the secure process will initiate the Stage 1 boot process.
+
+3.  The Stage 1 boot process will perform processor initialization, and
+    then initiate the Stage 2 boot process.
+
+4.  The Stage 2 boot process will initiate the Secure Loader, which will
+    process any customer specific customizations (e.g. front panel
+    of ECU, USB based image updates, etc).
+
+5.  The Secure Loader will check to determine if there are any updates
+    to be processed. If the update settings indicate that an upgrade
+    should occur then the Secure Loader will will determine the correct
+    action based on the nature of the upgrades:
+
+    a.  If the Secure Loader determines that an upgrade was performed
+        (or attempted), it will initiate the reboot process.
+
+    b.  If no upgrades were processed: then the Secure Loader will pass
+        control back to the Stage 2 boot process for further processing
+
+6.  The Stage 2 boot process will continue with the boot process, by
+    performing a verification of the kernel image prior to the load of
+    that image
+
+    a.  If the kernel image verification fails, the Stage 2 boot loader
+        will initiate a reboot
+
+8.  The Stage 2 boot loader will load the successfully verified kernel
+    image and boot the linux OS
+
+9.  The booted Linux OS will perform the normal Linux init sequence
+
+10. The Linux init process will start the required applications and
+    services as described in the init process and present on the rootfs.
 
 Requirements
 ============
 
   For the purposes of reference and explanation, we are providing guidance
   on how to configure an embedded device that runs with a linux 3. 10.17
-  Linux kernel.  
+  Linux kernel, and includes the use of U-Boot as the *Stage 2*
```

Comment:
You are right. We have only looked at U-Boot for the sake of this document, mostly because of the verified boot capability.